From 084c04cb195c72e3b51afd26527e0e042bb2ec3d Mon Sep 17 00:00:00 2001
From: Ankit Jain <ankitja@vmware.com>
Date: Mon, 7 Sep 2020 12:37:17 +0000
Subject: [PATCH] librepo: Fix the key string parsing

- break the loop  if last char of key string is '_'
- Use Case:
  Let say: releasever = 1.0
           basearch = x86_64
  String: $releasever_$basearch
  - Without fix:
   - parsed as $releasever_x86_64
  - With fix:
   - parsed as 1.0_x86_64

Signed-off-by: Ankit Jain <ankitja@vmware.com>
---
 librepo/url_substitution.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/librepo/url_substitution.c b/librepo/url_substitution.c
index 69d25bb..da1b0d5 100644
--- a/librepo/url_substitution.c
+++ b/librepo/url_substitution.c
@@ -124,7 +124,7 @@ lr_url_substitute(const char *url, LrUrlVars *list)
                 bracket = FALSE;
             }
             const char *varname = cur;
-            for (; isalnum(*cur) || *cur == '_'; ++cur);
+            for (; isalnum(*cur) || (*cur == '_' && isalnum(*(cur + 1))); ++cur);
             if (cur != varname && (!bracket || *cur == '}')) {
                 for (LrUrlVars *elem = list; elem; elem = g_slist_next(elem)) {
                     LrVar *var_val = elem->data;
-- 
2.23.3

